lib/_http_client.js

```diff
@@ -133,10 +133,12 @@ function ClientRequest(options, cb) {
   if (self.socketPath) {
     self._last = true;
     self.shouldKeepAlive = false;
-    const optionsPath = {
-      path: self.socketPath
+    const createConnectionOptions = {
+      path: self.socketPath,
+      socketTimeout: options.socketTimeout
```

Comment:
The reason I bring it up is that passing an option named `socketTimeout` to the default `net.Socket` constructor seems redundant. Even though there are other 'socket' references when it comes to http.Agent, the fewer the better IMHO since third party Agent instances could be used which may not use sockets at all.